apps/frontend/src/app/5_pages/ConvertPage/ConvertPage.tsx

```diff
@@ -21,78 +21,98 @@ import {
-const maximumPricePlaceholder = 30000;
+
+const tokensToOptions = (
+  addresses: string[],
+  callback: (options: SelectOption<SupportedTokens>[]) => void,
+) =>
+  Promise.all(
+    addresses.map(address => smartRouter.getTokenDetails(address)),
+  ).then(tokens =>
+    callback(
+      tokens.map(token => ({
+        value: token.symbol,
+        label: (
+          <AssetRenderer
+            showAssetLogo
+            asset={token.symbol}
+            assetClassName="font-medium"
+          />
+        ),
+      })),
+    ),
+  );
 
 const ConvertPage: FC = () => {
   const { account } = useAccount();
+  const [slippageTolerance, setSlippageTolerance] = useState('0.5');
 
   const defaultSourceToken = useGetDefaultSourceToken();
 
+  const [priceInQuote, setPriceQuote] = useState(false);
   const [amountInput, setAmount, amount] = useAmountInput('');
+  const [quote, setQuote] = useState('');
+  const [route, setRoute] = useState<SwapRoute | undefined>();
 
   const [sourceToken, setSourceToken] =
     useState<SupportedTokens>(defaultSourceToken);
 
-  const { checkMaintenance, States } = useMaintenance();
-  const convertLocked = checkMaintenance(States.ZERO_CONVERT);
-  const dllrLocked = checkMaintenance(States.ZERO_DLLR);
   const [showAdvancedSettings, setShowAdvancedSettings] = useState(false);
 
-  const tokenOptions = useMemo(
-    () =>
-      SupportedTokenList.filter(item =>
-        allowedTokens.includes(item.symbol),
-      ).map(token => ({
-        value: token.symbol,
-        label: (
-          <AssetRenderer
-            showAssetLogo
-            asset={token.symbol}
-            assetClassName="font-medium"
-          />
-        ),
-      })),
-    [],
-  );
+  const [tokenOptions, setTokenOptions] = useState<
+    SelectOption<SupportedTokens>[]
+  >([]);
 
-  const tokenOptionsWithoutSourceToken = useMemo(
-    () => tokenOptions.filter(item => item.value !== sourceToken),
-    [sourceToken, tokenOptions],
-  );
+  const [destinationTokenOptions, setDestinationTokenOptions] = useState<
+    SelectOption<SupportedTokens>[]
+  >([]);
 
-  const [destinationTokenOptions, setDestinationTokenOptions] = useState(
-    tokenOptionsWithoutSourceToken,
-  );
+  useEffect(() => {
+    smartRouter
+      .getEntries()
+      .then(tokens => tokensToOptions(tokens, setTokenOptions));
+  }, []);
 
   useEffect(() => {
-    setDestinationTokenOptions(tokenOptionsWithoutSourceToken);
-  }, [tokenOptionsWithoutSourceToken]);
+    (async () => {
+      const sourceTokenDetails = await getTokenDetails(
+        sourceToken,
+        defaultChainId,
+      );
+
+      smartRouter
+        .getDestination(sourceTokenDetails.address)
+        .then(tokens => tokensToOptions(tokens, setDestinationTokenOptions));
+    })();
+  }, [sourceToken]);
 
   const [destinationToken, setDestinationToken] = useState<
     SupportedTokens | ''
   >('');
+  const weiAmount = useMemo(() => toWei(amount), [amount]);
```

Comment:
maybe you can use `useWeiAmountInput` to get amount in wei. Instead of `useAmountInput` in line 76 for example